### DISCUSSION TITLE
label bind_text and backward loop

### DISCUSSION THREAD

**Discussion by marcin-koziol** [zauberzeug/nicegui#2202] [Q&A] [answered]
### Question

is it normal that backward is executed continuously (in a loop) constantly, even if there is no client?

here's a minimal example:

```python

data = {'name': 'Bob', 'age': 17}

def xd(v):
    print("greetings from backward")
    return f"Name == {v}"

ui.label().bind_text_from(data, 'name', backward=lambda n: xd(n))
ui.label().bind_text_from(data, 'age', backward=lambda a: f'Age: {a}')

ui.button('Turn 18', on_click=lambda: data.update(age=18))

ui.run(title='xDDDD 2137', dark=None, storage_secret='$@!#%@#Ewfasdf#', show=False, language='pl')

```

IMO, the backward funcion should be executed only when the value is changed, right?

**Comment by falkoschindler** [comment, marked as answer]
Hi @marcin-koziol,

Yes, that's normal. The binding loop is checking if the label text needs to be updated. For this to decide it needs to compute the new text, i.e. use the `backward` conversion function. Therefore these functions should be fast.

And you're right, in theory we could check if the source value changed. But to do so we would need to store a copy of all input values, which we don't do at the moment. It is certainly a tradeoff between storage, code complexity/reliability and performance.

This reminds me that we need to extend the documentation about binding. There are quite some details unmentioned.
